src/osd/OSD.cc

```diff
@@ -896,7 +898,7 @@ void OSDService::send_pg_temp()
   MOSDPGTemp *m = new MOSDPGTemp(osdmap->get_epoch());
   m->pg_temp = pg_temp_wanted;
   monc->send_mon_message(m);
-  _sent_pg_temp();
+  pg_temp_pending.swap(pg_temp_wanted);
```

Comment:
@tchaikov  It is relate to the function requeue_pg_temp .Just from the local send_pg_temp it is not equal. But I change it back here, to make it clear to read.
 service.requeue_pg_temp();
  service.send_pg_temp();